addons/SpellBook/SpellBook.lua

```diff
@@ -0,0 +1,444 @@
+            main_level = nil
+        end
+        if sub_level and sub_level > player.sub_job_level then
+            sub_level = nil
+        end
+
+        if main_level and sub_level then
+            spell_level = math.min(main_level, sub_level)
+        else
+            spell_level = spell_level or main_level or sub_level
+        end
+
+        if spell_level and spell.type ~= 'Trust' and
+            not player_spells[spell_id] and not spell.unlearnable then
+
+            missing_spells:append(format_spell(spell))
+            spell_count = spell_count + 1
+        end
+    end
+
+    if not missing_spells:empty() then
+        if main_job_jp > 0 then
+            if player.sub_job then
+                windower.add_to_chat(7, string.format('Showing learnable spells for %s%d with %d spent job points and %s%d.',
+                    player.main_job, player.main_job_level,
+                    player.job_points[player.main_job:lower()].jp_spent,
+                    player.sub_job, player.sub_job_level))
+            else
+                windower.add_to_chat(7, string.format('Showing learnable spells for %s%d with %d spent job points.',
+                    player.main_job, player.main_job_level,
+                    player.job_points[player.main_job:lower()].jp_spent))
+            end
+        else
+            if player.sub_job then
+                windower.add_to_chat(7, string.format('Showing learnable spells for %s%d/%s%d.',
+                    player.main_job, player.main_job_level,
+                    player.sub_job, player.sub_job_level))
+            else
+                windower.add_to_chat(7, string.format('Showing learnable spells for %s%d.',
+                    player.main_job, player.main_job_level))
+            end
+        end
+
+        missing_spells:sort()
+        for _,spell in ipairs(missing_spells) do
+            windower.add_to_chat(7, spell)
+        end
+        if player.sub_job then
+            windower.add_to_chat(7, string.format('List Complete. You are missing %d learnable spells for %s%d/%s%d.',
+                spell_count,
+                player.main_job, player.main_job_level,
+                player.sub_job, player.sub_job_level))
+        else
+            windower.add_to_chat(7, string.format('List Complete. You are missing %d learnable spells for %s%d.',
+                spell_count,
+                player.main_job, player.main_job_level))
+        end
+    else
+        if player.sub_job then
+            windower.add_to_chat(7,  string.format('Congratulations! You know all learnable spells for %s%d/%s%d!',
+                player.main_job, player.main_job_level,
+                player.sub_job, player.sub_job_level))
+        else
+            windower.add_to_chat(7,  string.format('Congratulations! You know all learnable spells for %s%d!',
+                player.main_job, player.main_job_level))
+        end
+    end
+end
+
+--[[
+Copyright © 2018, John S Hobart
```

Comment:
I assume that is you? Just checking, not that you forgot to change the name, it happens...

Reply:
Yeah that is me. heh